bundles/io/org.openhab.io.transport.mqtt/src/main/java/org/openhab/io/transport/mqtt/MqttService.java

```diff
@@ -91,7 +91,13 @@ public void updated(Dictionary<String, ?> properties) throws ConfigurationExcept
             } else if (property.equals("async")) {
                 conn.setAsync(Boolean.parseBoolean(value));
             } else if (property.equals("clientId")) {
+                if (value.length() > 65535) {
+                    logger.warn("clientId must be less than 65536 characters long");
+                    throw new ConfigurationException("clientId", "clientId must be less than 65536 characters long");
+                }
                 conn.setClientId(value);
+            } else if (property.equals("allowLongerClientIds")) {
+                conn.setAllowLongerClientIds(Boolean.parseBoolean(value));
```

Comment:
That was the problem. I had tried removing the built-in jar, but I put the new jar back in the same directory instead of in the addons folder. Thanks!